docs/spec_draft.md

```diff
@@ -1406,6 +1407,69 @@ and produces a `result` tensor. More formally,
 ```
 
 [Back to Ops](#index-of-ops)
+
+## stablehlo.rng_bit_generator
+
+### Semantics
+
+Returns an `output` filled with uniform random data and an updated
+`output_state` using the `rng_algorithm` given `initial_state`. The output is
+guaranteed to be deterministic from the initial state, but it is not guaranteed
+to be deterministic between backends and different compiler versions.
+
+`initial_state` is the initial state of the current random number generation.
+The initial state, required shape, and valid values are dependent on the
```

Comment:
I think by `required shape` they mean the shape of the initial state. Because the output shape is provided by the user only.
Having said that we can remove the entire sentence*** because we already talked about the  shape dependence on algorithm in the constants section. 

*** Before removing please also look into the following:

`valid values are dependent on the algorithm`: 
Can we know what do they mean by "valid values". IOW, what qualifies value to be valid Vs  invalid ?

Reply:
By valid values, it means the element type since THREE_FRY and PHILOX algorithms requires ui64 `initial_state` and outputs ui64 `output_state` and outputs either u32 or u64 `output` (aka, no other precision or element types are allowed implicitly in the verifier).

There is no explicit check for this since the DEFAULT prng algorithm may support *uniform* (it's hard to produce uniform floating-points since many bit representations can represent NaNs, for example) floating-point prng, which seems to be a very niche [special case](https://electronics.stackexchange.com/questions/586009/how-is-a-floating-point-prng-designed) though.

I think the use case of this op is to generate uniformly random bits, not random numbers.